langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/PartsMapper.java

```diff
@@ -86,7 +93,12 @@ static List<Part> map(ChatMessage message) {
                 try {
                     JsonFormat.parser().merge(functionResponseTextAsMap, structBuilder);
                 } catch (InvalidProtocolBufferException e2) {
-                    throw new RuntimeException(e);
+                    String functionResponseTextWithQuotesAsMap = "{\"result\":" + quoted(functionResponseText) + "}";
```

Comment:
Can we predict if response text should be in a map and/or quoted? I would avoid using exceptions to control the flow

Reply:
I followed the _pattern_ that was used with the nested exception control flow.
I'm not quite sure what is the best approach here, as a function can return even raw text, not necessarily JSON.

My understanding is that 1) if it's JSON, it works fine, 2) if it's some value like a number, a boolean, it can be returned in a wrapped JSON object with a `result` key, but 3) if it's some text, it has to be quoted. Hence the three levels of nesting.
We have to parse somehow to know in which case we are.

But if you have a better idea, I'm interested.

Reply:
I see. Maybe we should always quote the `functionResponseText` to avoid the third level of nesting? If it is number or boolean, I guess it is still OK to quote it. But it would be nice to understand why this must JSON at all.

Reply:
I wanted to do that initially, but decided against it, as it means we would always wrap even basic types like numbers, booleans, or null, into strings.